client/client.go

```diff
@@ -12,34 +12,49 @@ import (
 )
 
 var (
-	app   = kingpin.New("client", "send messages to a server")
-	host  = app.Flag("host", "host to connect to").Required().String()
-	send  = app.Flag("message", "send a request to server").Required().Strings()
-	token = app.Flag("token", "authentication token").Required().String()
+	app = kingpin.New("client", "connect to server")
+
+	send    = app.Command("send", "send message to server")
+	host    = send.Flag("host", "host to connect to").Required().String()
+	message = send.Flag("message", "send a request to server").Required().Strings()
+	token   = app.Flag("token", "authentication token").Required().String()
 )
 
 func main() {
-	kingpin.MustParse(app.Parse(os.Args[1:]))
-	message := strings.Join(*send, " ")
-	reader := strings.NewReader(message)
-
-	req, err := http.NewRequest("POST", *host+"/echo", reader)
-
-	// add token header to the req
-	req.Header.Add("Token", *token)
-
-	client := &http.Client{}
-	resp, err := client.Do(req)
-
-	if err != nil {
-		log.Println(err)
-	}
-
-	defer resp.Body.Close()
-	bodyBytes, err := ioutil.ReadAll(resp.Body)
-	serverOutput := string(bodyBytes)
-	if serverOutput != message {
-		fmt.Println(*host + " did not echo your request, oops! got:")
+	switch kingpin.MustParse(app.Parse(os.Args[1:])) {
+
+	case send.FullCommand():
+		requestMessage := strings.Join(*message, " ")
+		reader := strings.NewReader(requestMessage)
+		endpoint := *host + "/echo"
+		req, err := http.NewRequest("POST", endpoint, reader)
+		if err != nil {
+			fmt.Println("request could not be initiated: ", err)
+			return 
+		}
+		req.Header.Add("Token", *token)
+		client := &http.Client{}
+		resp, err := client.Do(req)
+		if err != nil {
+			fmt.Println("response error: ", err)
+			return 
+		}
+
+		defer resp.Body.Close()
+		if resp.StatusCode != http.StatusOK {
+			fmt.Println("Not OK HTTP status:", resp.StatusCode)
+			return
+		}
+
+		bodyBytes, err := ioutil.ReadAll(resp.Body)
+		if err != nil {
+			log.Fatalf("could not read body: %v", err)
```

Comment:
Why l`og.Fatalf` here, but `fmt.Println` above?

Reply:
I wanted the program to stop if the client couldn't read the response, but I suppose a more appropriate solution would be to use fmt and return